src/main/java/com/exactpro/th2/conn/ServiceListener.java

```diff
@@ -76,20 +77,20 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(ServiceListener.class);
 
     private final Map<Direction, AtomicLong> directionToSequence;
-    private final Builder idBuilder;
+    private final Supplier<Builder> idBuilder;
     private final String sessionAlias;
     private final Subscriber<ConnectivityMessage> subscriber;
     private final EventDispatcher eventDispatcher;
 
     public ServiceListener(
             Map<Direction, AtomicLong> directionToSequence,
-            Builder idBuilder,
+            Supplier<Builder> idBuilder,
             Subscriber<ConnectivityMessage> subscriber,
             EventDispatcher eventDispatcher
     ) {
         this.directionToSequence = requireNonNull(directionToSequence, "Map direction to sequence counter can't be null");
         this.idBuilder = requireNonNull(idBuilder, "Message id builder can't be null");
-        this.sessionAlias = requireNonNull(idBuilder.getConnectionId().getSessionAlias(), "Session alias can't be null");
+        this.sessionAlias = requireNonNull(idBuilder.get().getConnectionId().getSessionAlias(), "Session alias can't be null");
```

Comment:
Maybe the sessionAlias is redundant

Reply:
I don't think so, it's used in many log mesages and here:
`DIRECTION_TO_COUNTER.get(direction).labels(sessionAlias).inc();`
`DIRECTION_TO_COUNTER` used for metrics `th2_conn_incoming_msg_quantity` and `th2_conn_outgoing_msg_quantity`.